### ISSUE TITLE
builds broken due to grpc 1.3.0

### ISSUE THREAD

**Issue by ofrobots** [GoogleCloudPlatform/cloud-trace-nodejs#473]
Builds are failing. This is due to a new version of grpc recently released this week.

```
❯ mocha test/test-grpc-context.js -g "preserve context for server requests"


  express + grpc1
Server#addProtoService is deprecated. Use addService instead
TypeError: Cannot read property 'n' of undefined
    at Type._nodetest_TestRequest$fromObject (eval at eof (/Users/ofrobots/src/cloud-trace/test/plugins/fixtures/grpc1/node_modules/@protobufjs/codegen/index.js:103:25), <anonymous>:7:6)
    at Type.fromObject (/Users/ofrobots/src/cloud-trace/test/plugins/fixtures/grpc1/node_modules/protobufjs/src/type.js:479:25)
    at serialize (/Users/ofrobots/src/cloud-trace/test/plugins/fixtures/grpc1/node_modules/grpc/src/node/src/protobuf_js_6_common.js:80:23)
    at Client.makeServerStreamRequest (/Users/ofrobots/src/cloud-trace/test/plugins/fixtures/grpc1/node_modules/grpc/src/node/src/client.js:565:19)
    at Client.clientMethodTrace [as testServerStream] (/Users/ofrobots/src/cloud-trace/src/plugins/plugin-grpc.js:117:25)
    at /Users/ofrobots/src/cloud-trace/test/test-grpc-context.js:87:29
    at Layer.handle [as handle_request] (/Users/ofrobots/src/cloud-trace/test/plugins/fixtures/express4/node_modules/express/lib/router/layer.js:95:5)
    at next (/Users/ofrobots/src/cloud-trace/test/plugins/fixtures/express4/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/Users/ofrobots/src/cloud-trace/test/plugins/fixtures/express4/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/Users/ofrobots/src/cloud-trace/test/plugins/fixtures/express4/node_modules/express/lib/router/layer.js:95:5)
    at /Users/ofrobots/src/cloud-trace/test/plugins/fixtures/express4/node_modules/express/lib/router/index.js:281:22
    at Function.process_params (/Users/ofrobots/src/cloud-trace/test/plugins/fixtures/express4/node_modules/express/lib/router/index.js:335:12)
    at next (/Users/ofrobots/src/cloud-trace/test/plugins/fixtures/express4/node_modules/express/lib/router/index.js:275:10)
    at /Users/ofrobots/src/cloud-trace/src/plugins/plugin-express.js:77:7
    at /Users/ofrobots/src/cloud-trace/src/trace-api.js:160:12
    at /Users/ofrobots/src/cloud-trace/node_modules/continuation-local-storage/context.js:65:13
    at Namespace.run (/Users/ofrobots/src/cloud-trace/node_modules/continuation-local-storage/context.js:48:5)
    at Namespace.runAndReturn (/Users/ofrobots/src/cloud-trace/node_modules/continuation-local-storage/context.js:64:8)
    at TraceApiImplementation.runInRootSpan (/Users/ofrobots/src/cloud-trace/src/trace-api.js:157:32)
    at TraceApi.runInRootSpan (/Users/ofrobots/src/cloud-trace/src/trace-api.js:290:19)
    at middleware (/Users/ofrobots/src/cloud-trace/src/plugins/plugin-express.js:42:9)
    at Layer.handle [as handle_request] (/Users/ofrobots/src/cloud-trace/test/plugins/fixtures/express4/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/Users/ofrobots/src/cloud-trace/test/plugins/fixtures/express4/node_modules/express/lib/router/index.js:317:13)
    at /Users/ofrobots/src/cloud-trace/test/plugins/fixtures/express4/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/Users/ofrobots/src/cloud-trace/test/plugins/fixtures/express4/node_modules/express/lib/router/index.js:335:12)
    at next (/Users/ofrobots/src/cloud-trace/test/plugins/fixtures/express4/node_modules/express/lib/router/index.js:275:10)
    at expressInit (/Users/ofrobots/src/cloud-trace/test/plugins/fixtures/express4/node_modules/express/lib/middleware/init.js:40:5)
    at Layer.handle [as handle_request] (/Users/ofrobots/src/cloud-trace/test/plugins/fixtures/express4/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/Users/ofrobots/src/cloud-trace/test/plugins/fixtures/express4/node_modules/express/lib/router/index.js:317:13)
    at /Users/ofrobots/src/cloud-trace/test/plugins/fixtures/express4/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/Users/ofrobots/src/cloud-trace/test/plugins/fixtures/express4/node_modules/express/lib/router/index.js:335:12)
    at next (/Users/ofrobots/src/cloud-trace/test/plugins/fixtures/express4/node_modules/express/lib/router/index.js:275:10)
    at query (/Users/ofrobots/src/cloud-trace/test/plugins/fixtures/express4/node_modules/express/lib/middleware/query.js:44:5)
    at Layer.handle [as handle_request] (/Users/ofrobots/src/cloud-trace/test/plugins/fixtures/express4/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/Users/ofrobots/src/cloud-trace/test/plugins/fixtures/express4/node_modules/express/lib/router/index.js:317:13)
    at /Users/ofrobots/src/cloud-trace/test/plugins/fixtures/express4/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/Users/ofrobots/src/cloud-trace/test/plugins/fixtures/express4/node_modules/express/lib/router/index.js:335:12)
    at next (/Users/ofrobots/src/cloud-trace/test/plugins/fixtures/express4/node_modules/express/lib/router/index.js:275:10)
    at Function.handle (/Users/ofrobots/src/cloud-trace/test/plugins/fixtures/express4/node_modules/express/lib/router/index.js:174:3)
    at Function.handle (/Users/ofrobots/src/cloud-trace/test/plugins/fixtures/express4/node_modules/express/lib/application.js:174:10)
    at Server.app (/Users/ofrobots/src/cloud-trace/test/plugins/fixtures/express4/node_modules/express/lib/express.js:38:9)
    at emitTwo (events.js:106:13)
    at Server.emit (events.js:194:7)
    at parserOnIncoming (_http_server.js:565:12)
    at HTTPParser.parserOnHeadersComplete (_http_common.js:99:23)

    1) grpc should preserve context for server requests


  0 passing (233ms)
  1 failing

  1) express + grpc1 grpc should preserve context for server requests:

      Uncaught AssertionError: 1 === 2
      + expected - actual

      -1
      +2

      at ClientRequest.<anonymous> (test/test-grpc-context.js:193:16)
      at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:522:21)
      at HTTPParser.parserOnHeadersComplete (_http_common.js:99:23)
      at Socket.socketOnData (_http_client.js:411:20)
      at readableAddChunk (_stream_readable.js:178:18)
      at Socket.Readable.push (_stream_readable.js:136:10)
      at TCP.onread [as _originalOnread] (net.js:559:20)
      at TCP.onread (node_modules/async-listener/glue.js:188:31)
```

Switching to grpc 1.2.x makes the test pass.

/cc @murgatroid99 for awareness.

**Comment by ofrobots** [comment]
It seems that the following client.js code no longer handles `argument` being `undefined`:

```js
  /**
   * Make a server stream request with this method on the given channel with the
   * given argument, etc.
   * @this {SurfaceClient} Client object. Must have a channel member.
   * @param {*} argument The argument to the call. Should be serializable with
   *     serialize
   * @param {Metadata=} metadata Array of metadata key/value pairs to add to the
   *     call
   * @param {Object} options Options map
   * @return {EventEmitter} An event emitter for stream related events
   */
  function makeServerStreamRequest(argument, metadata, options) {
    /* jshint validthis: true */
    /* While the arguments are listed in the function signature, those variables
     * are not used directly. Instead, ArgueJS processes the arguments
     * object. */
    var args = arguejs({argument: null, metadata: [Metadata, new Metadata()],
                        options: [Object]}, arguments);
    var call = getCall(this.$channel, method, args.options);
    metadata = args.metadata.clone();
    var stream = new ClientReadableStream(call, deserialize);
    var start_batch = {};
    var message = serialize(args.argument);
```

`serialize` throws when called with undefined.

**Comment by ofrobots** [comment]
@murgatroid99 this worksaround the problem here: https://github.com/GoogleCloudPlatform/cloud-trace-nodejs/pull/474. Is this an intentional change in gRPC, or is it a bug? Should I open a bug on your issue tracker?

**Comment by murgatroid99** [comment]
I previously had no idea that `undefined` was ever an accepted argument. Please do open an issue; this should be easy enough to fix.

**Comment by ofrobots** [comment]
opened https://github.com/grpc/grpc/issues/10952

**Comment by murgatroid99** [comment]
I am switching gRPC's dependency on Protobuf.js back to version 5 to fix this.

**Comment by murgatroid99** [comment]
I have published grpc@1.3.1-pre1, which switches the dependency on Protobuf.js back to version 5. Can you verify that this library works with that?
